contextbasedrestrictionsv1/context_based_restrictions_v1.go

```diff
@@ -1839,20 +1868,37 @@ type ReplaceRuleOptions struct {
 	// The resources this rule apply to.
 	Resources []Resource `json:"resources,omitempty"`
 
+	// The rule enforcement mode:
+	//  * `enabled` - The restrictions are enforced and reported. This is the default.
+	//  * `disabled` - The restrictions are disabled. Nothing is enforced or reported.
+	//  * `report` - The restrictions are evaluated and reported, but not enforced.
+	EnforcementMode *string `json:"enforcement_mode,omitempty"`
+
 	// The supplied or generated value of this header is logged for a request and repeated in a response header for the
 	// corresponding response. The same value is used for downstream requests and retries of those requests. If a value of
 	// this headers is not supplied in a request, the service generates a random (version 4) UUID.
-	XCorrelationID *string `json:"-"`
+	XCorrelationID *string `json:"X-Correlation-Id,omitempty"`
 
 	// The `Transaction-Id` header behaves as the `X-Correlation-Id` header. It is supported for backward compatibility
 	// with other IBM platform services that support the `Transaction-Id` header only. If both `X-Correlation-Id` and
 	// `Transaction-Id` are provided, `X-Correlation-Id` has the precedence over `Transaction-Id`.
-	TransactionID *string `json:"-"`
+	TransactionID *string `json:"Transaction-Id,omitempty"`
 
 	// Allows users to set headers on API requests
 	Headers map[string]string
 }
 
+// Constants associated with the ReplaceRuleOptions.EnforcementMode property.
+// The rule enforcement mode:
+//  * `enabled` - The restrictions are enforced and reported. This is the default.
+//  * `disabled` - The restrictions are disabled. Nothing is enforced or reported.
+//  * `report` - The restrictions are evaluated and reported, but not enforced.
+const (
+	ReplaceRuleOptionsEnforcementModeDisabledConst = "disabled"
+	ReplaceRuleOptionsEnforcementModeEnabledConst  = "enabled"
+	ReplaceRuleOptionsEnforcementModeReportConst   = "report"
+)
```

Comment:
those are auto-generated

Reply:
These are generated as a result of the fact that the corresponding schema defines an enum.  Why delete the const definitions here???

Reply:
@padamstx if everything is populated from the swagger it seems that we should check our swagger to make sure we have 1 definition that gets reused then. @zhenwan can you take a look into making these changes?

Reply:
The duplicates might be generated due to the way in which the options model structs are synthesized.  The generator will "explode" body vars which means that when representing a request body within the options model struct, the generator will essentially copy each of the request body's top-level properties (assume the request body is defined as an "object").  This might result in duplicates because the options model struct AND the original request body schema would now both define the "enforcement_mode" property which (I assume) contains the enum definition.
Viewing the duplicates through that lens, it makes sense that we're seeing the duplicate definitions.  My suggestion would be to just leave the duplicate const definitions there since any changes to the generated code will need to be applied every time you re-gen the service.